packages/http-client-csharp/emitter/src/emitter.ts

```diff
@@ -155,20 +155,17 @@ export async function $onEmit(context: EmitContext<NetEmitterOptions>) {
             debugFlag,
           ],
           { stdio: "inherit" }
-        )
-          .then(() => {
-            if (!options["save-inputs"]) {
-              // delete
-              deleteFile(resolvePath(outputFolder, tspOutputFileName));
-              deleteFile(resolvePath(outputFolder, configurationFileName));
-            }
-          })
-          .catch((error: any) => {
-            if (error.message) Logger.getInstance().info(error.message);
-            if (error.stderr) Logger.getInstance().error(error.stderr);
-            if (error.stdout) Logger.getInstance().verbose(error.stdout);
-            throw error;
-          });
+        );
+        if (result.exitCode !== 0) {
+          if (result.stderr) Logger.getInstance().error(result.stderr);
+          if (result.stdout) Logger.getInstance().verbose(result.stdout);
+          throw new Error(`Failed to generate SDK. Exit code: ${result.exitCode}`);
```

Comment:
is this an expected scenario that this would fail or will it be representative of a bug?

Doing this will report to the user if it happens that there is a bug in your emitter.

```
ExternalError: Emitter "@azure-tools/typespec-autorest" crashed! This is a bug.
Please file an issue at https://github.com/Azure/typespec-azure/issues

Error: blah
    at getSdkHttpResponseAndExceptions (https://cadlplayground.z22.web.core.windows.net/cadl-azure/libs/@azure-tools/typespec-client-generator-core/index.js:2912:27)
    at getSdkHttpOperation (https://cadlplayground.z22.web.core.windows.net/cadl-azure/libs/@azure-tools/typespec-client-generator-core/index.js:2668:56)
    at getSdkServiceOperation (https://cadlplayground.z22.web.core.windows.net/cadl-azure/libs/@azure-tools/typespec-client-generator-core/index.js:3009:51)
    at getSdkBasicServiceMethod (https://cadlplayground.z22.web.core.windows.net/cadl-azure/libs/@azure-tools/typespec-client-generator-core/index.js:3131:47)
    at getSdkServiceMethod (https://cadlplayground.z22.web.core.windows.net/cadl-azure/libs/@azure-tools/typespec-client-generator-core/index.js:3164:12)
    at getSdkMethods (https://cadlplayground.z22.web.core.windows.net/cadl-azure/libs/@azure-tools/typespec-client-generator-core/index.js:3236:38)
    at createSdkClientType (https://cadlplayground.z22.web.core.windows.net/cadl-azure/libs/@azure-tools/typespec-client-generator-core/index.js:3344:38)
    at getSdkPackage (https://cadlplayground.z22.web.core.windows.net/cadl-azure/libs/@azure-tools/typespec-client-generator-core/index.js:3366:9)
    at createSdkContext (https://cadlplayground.z22.web.core.windows.net/cadl-azure/libs/@azure-tools/typespec-client-generator-core/index.js:3772:42)
    at Object.$onEmit [as emitFunction] (https://cadlplayground.z22.web.core.windows.net/cadl-azure/libs/@azure-tools/typespec-autorest/index.js:613:28)

--------------------------------------------------
Library Version                0.41.1
TypeSpec Compiler Version      0.55.0
--------------------------------------------------
```

Reply:
It is not expected that the dotnet call exits with anything other than 0.  If this ever happens it should be reported as a bug.